mobile/src/main/java/org/openhab/habdroid/ui/activity/PageConnectionHolderFragment.java

```diff
@@ -69,6 +69,11 @@
          * @param widget Updated widget
          */
         void onWidgetUpdated(OpenHABWidget widget);
+
+        /**
+         * Let parent know about a failure during the load of data.
+         */
+        void onLoadFailure(int statusCode, Headers headers, byte[] responseBody, Throwable error);
```

Comment:
I'd omit the unneeded headers and responseBody parameters from this callback.

Reply:
+1